src/Routing/Setter.purs

```diff
@@ -10,6 +10,14 @@ function setHash(hash) {
 }
 """ :: forall e. String -> Eff e Unit
 
+foreign import getHash """
+function getHash() {
+  return document.location.hash.replace(/^[^#]*#/g, "");
+}
+""" :: forall e. Eff e String
+
+class RouteModifier a where
```

Comment:
Why do we need a type class for this? Is it useful to have only a single modifier per type? It seems like it could just be a function down below for modifyRoute.